### ISSUE TITLE
Synchronized cloning inside PebbleView

### ISSUE THREAD

**Issue by hectorlf** [PebbleTemplates/pebble-spring3#8]
Hey, @ebussieres, I'm adding some things to pebble-spring3 that I need and I'd like to ask a question. Why the synchronized cloning and caching of Beans maps? I'm failing to understand the reasoning behind it :-/

**Comment by ebussieres** [comment]
This part is based on thymeleaf (https://github.com/thymeleaf/thymeleaf-spring/blob/2.1-dev/thymeleaf-spring4/src/main/java/org/thymeleaf/spring4/context/SpringWebContext.java)

Cloning is mainly for BindingResult to work or anything else added by spring during runtime. You don't want to share them with all threads. There's a unit test, remove the cloning part and you'll see that the test won't work.

Hope this help :)

**Comment by hectorlf** [comment]
Oh, sorry, I focused on the wrong side of the question. I understand that the map should not be shared among threads, but why cache the maps in the first place? Appcontext is "mostly read-only", the Beans object is immutable & cheap to create and cloning a Map is the same as instantiating a new one. I fail to see the benefit of caching a map just to clone it in every request, moreso when it introduces a synchronized block that is going to lock all threads on the same object :S

If you don't mind, I'd like to simplify the map handling and remove all forms of synchonization.

**Comment by ebussieres** [comment]
I totally agree with you I think it could be simplified. As long as the unit test work, I've got no problem with that.

Somethink like
```
private static Map<String, Object> addSpringBeansVariable(Map<String, ?> model, ApplicationContext appctx) {
        Map<String, Object> variableMap = new HashMap<>();
        variableMap.put(BEANS_VARIABLE_NAME, new Beans(appctx));

        if (model != null) {
            variableMap.putAll(model);
        }

        return variableMap;
}
```

**Comment by hectorlf** [comment]
Yeah, perfect! Would you like to do the pull request? (YCYPR: Your Code Your Pull Request, LOL)

**Comment by ebussieres** [comment]
Sure i'll do it :D

**Comment by mbosecke** [comment]
Hey, @ebussieres , since you've contributed so much to pebble-spring3 would you be interested in being the primary maintainer for this project? I've recently created a PebbleTemplates organization, which I can invite you to join and you'll have full write access to this particular repository. @hectorlf  is the only other member of the organization so far and he's currently maintaining the pebble-spring-boot-starter project.

**Comment by ebussieres** [comment]
@mbosecke Yeah sure I'm pleased to join the team.

**Comment by mbosecke** [comment]
@ebussieres and @hectorlf , if you guys don't mind can you send me an email at mitchellbosecke@gmail.com so that I have your email addresses? I'd like to communicate in private and github doesn't provide a way to do so.

**Comment by ebussieres** [comment]
Fixed via commit 878f787

**Comment by hectorlf** [comment]
Thank you!!
